app/posts/creating-successful-govuk-tools.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Building useful tools
+description: Thoughts on what it takes to make something other teams will use
+date: 2024-01-16
+author:
+  name: Peter Yates
+  url: https://github.com/peteryates
+image:
+  src: /assets/posts/creating-successful-govuk-tools/illustration.png
+  alt: Illustration of a red ruby gem, with multiple colours emanating from it.
+  opengraphImage: true
+---
+
+I joined the Department for Education's digital division in 2019 to work on [Get School Experience](https://schoolexperience.education.gov.uk/). It's a small service that lets people apply to spend a few days in a school, hopefully giving them a better idea of whether or not teaching is the career for them.
+
+I remember how refreshing it felt to join an organisation where working collaboratively wasn't just encouraged, it was the default. It's even baked into the service standard:
+
+> If you develop your own patterns or components, share them publicly so others can use them.
+>
+> <cite>[The GOV.UK Service Standard, point 13.](https://www.gov.uk/service-manual/service-standard/point-13-use-common-standards-components-patterns)</cite>
+
+My first job on the project was to work out how we were going capture all the information from applicants. We knew we were going to be using the newly-released GOV.UK Design System, and because Rails was popular across government we thought there would be established tools we could lean on.
+
+Forms are the main part of any transactional service, and Rails makes creating forms really straightforward. There must be something out there that combines the two.
+
+Unfortunately, there wasn't. We had a really tight deadline so repurposing the [Ministry of Justice's GOV.UK Elements form builder](https://github.com/ministryofjustice/govuk_elements_form_builder) for the GOV.UK Design System. It worked for us but never felt quite right.
+
+But, I'd found a gap. Something I thought should exist but didn't. Something I was certain others would find useful. Something I thought I could do.
+
+I decided to have a go at building and sharing a form builder from scratch and spent my free time over the next month working on it. I announced it [via a very nervous-sounding YouTube screencast](https://www.youtube.com/watch?v=PhoFZ0qXAlA) and, thankfully, it was well received.
+
+It was my first time building and publishing a library before, but having used plenty I had a good idea of what to prioritise if I wanted to make it a success. For me at the time, success meant a other teams finding it useful enough to use. Here's my list:
```

Comment:
Maybe before you go into the list you could briefly summarise how successful your library has been? Feel like you almost undersell it here!

Reply:
I think I've done this now